tests/functional_tests/ffront_tests/test_unrollPowOp.py

```diff
@@ -0,0 +1,139 @@
+import pathlib
+
+from eve.pattern_matching import ObjectPattern
+from functional.ffront import common_types as ct, field_operator_ast as foast
+from functional.ffront.func_to_foast import UnrollPowerOp
+
+
+def _make_power_testee(pow_n: str):
+    _set_source_location = foast.SourceLocation(
+        line=106, column=12, source=str(pathlib.Path(__file__).resolve())
+    )
+
+    testee = foast.BinOp(
+        right=foast.Constant(
+            dtype=ct.ScalarType(kind=ct.ScalarKind.INT),
+            location=_set_source_location,
+            value=pow_n,
+            type=ct.ScalarType(kind=ct.ScalarKind.INT),
+        ),
+        left=foast.Name(
+            id="a",
+            type=ct.DeferredSymbolType(constraint=None),
+            location=_set_source_location,
+        ),
+        op=foast.BinaryOperator.POW,
+        location=_set_source_location,
+        type=ct.DeferredSymbolType(constraint=None),
+    )
+    return testee
+
+
+def test_power_1():
+
+    actual = UnrollPowerOp().visit(_make_power_testee("1"))
+
+    expected = ObjectPattern(foast.Name, id="a")
+
+    assert expected.match(actual)
+
+
+def test_power_2():
+
+    actual = UnrollPowerOp().visit(_make_power_testee("2"))
+
+    expected = ObjectPattern(
+        foast.BinOp,
+        right=ObjectPattern(foast.Name, id="a"),
+        op=foast.BinaryOperator.MULT,
+    )
+
+    assert expected.match(actual)
+
+
+def test_power_3():
+
+    actual = UnrollPowerOp().visit(_make_power_testee("3"))
+
+    expected = ObjectPattern(
+        foast.BinOp,
+        right=ObjectPattern(foast.Name, id="a"),
+        left=ObjectPattern(
+            foast.BinOp,
+            right=ObjectPattern(foast.Name, id="a"),
+            left=ObjectPattern(foast.Name, id="a"),
+            op=foast.BinaryOperator.MULT,
+        ),
+        op=foast.BinaryOperator.MULT,
+    )
+    assert expected.match(actual)
+
+
+def test_power_0():
+    try:
+        UnrollPowerOp().visit(_make_power_testee("0"))
+    except ValueError:
+        True
```

Comment:
See [here](https://github.com/GridTools/gt4py/blob/67d97b2e9cc4dbf4cdf52973d8a81b33927f1041/tests/functional_tests/ffront_tests/test_interface.py#L76) and for an example and [here](https://docs.pytest.org/en/6.2.x/assert.html#assertions-about-expected-exceptions) for pytest documentation on how to properly test for expected exceptions. As a site note, the `True` here has no effect beside circumventing a syntax error. The proper way to archieve this in Python is to use `pass`.